lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/WritableGroupCommDriver.cs

```diff
@@ -0,0 +1,268 @@
+                .Build();
+
+            return TangFactory.GetTang().NewConfigurationBuilder(serviceConfig)
+                .BindImplementation(
+                    GenericType<IObserver<WritableNsMessage<WritableGeneralGroupCommunicationMessage>>>.Class,
+                    GenericType<WritableGroupCommNetworkObserver>.Class)
+                .BindNamedParameter<NamingConfigurationOptions.NameServerAddress, string>(
+                    GenericType<NamingConfigurationOptions.NameServerAddress>.Class, 
+                    _nameServerAddr)
+                .BindNamedParameter<NamingConfigurationOptions.NameServerPort, int>(
+                    GenericType<NamingConfigurationOptions.NameServerPort>.Class, 
+                    _nameServerPort.ToString(CultureInfo.InvariantCulture))
+                .BindImplementation(GenericType<INameClient>.Class,
+                    GenericType<NameClient>.Class)
+                .Build();
+        }
+
+        /// <summary>
+        /// Get the configuration for a particular task.  
+        /// The task may belong to many Communication Groups, so each one is serialized
+        /// in the configuration as a SerializedGroupConfig.
+        /// The user must merge their part of task configuration (task id, task class)
+        /// with this returned Group Communication task configuration.
+        /// </summary>
+        /// <param name="taskId">The id of the task Configuration to generate</param>
+        /// <returns>The Group Communication task configuration with communication group and
+        /// operator configuration set.</returns>
+        public IConfiguration GetGroupCommTaskConfiguration(string taskId)
+        {
+            var confBuilder = TangFactory.GetTang().NewConfigurationBuilder();
+
+            foreach (ICommunicationGroupDriver commGroup in _commGroups.Values)
+            {
+                var taskConf = commGroup.GetGroupTaskConfiguration(taskId);
+                if (taskConf != null)
+                {
+                    confBuilder.BindSetEntry<GroupCommConfigurationOptions.SerializedGroupConfigs, string>(
+                        GenericType<GroupCommConfigurationOptions.SerializedGroupConfigs>.Class,
+                        _configSerializer.ToString(taskConf));
+                }
+            }
+
+            return confBuilder.Build();
+        }
+
+        /// <summary>
+        /// Get the service configuration required for running Group Communication on Reef tasks.
+        /// </summary>
+        /// <param name="config">User specified configuration to be passed to service</param>
+        /// <returns>The service configuration for the Reef tasks</returns>
+        public IConfiguration GetServiceConfiguration(params IConfiguration[] config)
+        {
+            var serviceConfig = GetServiceConfiguration();
+            var allConfigs = new List<IConfiguration>(config) {serviceConfig};
```

Comment:
So I am a little confused. The Merge function needs array of configurations to merge right or pass individual condigurations. My case is different since I have one array of configurations and one single configuration that I need to merge. How can I do that?